File: NutzMahn/Arrays & Hashing/128_longest_consecutive_sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0){
            return 0;
        }else if(nums.length == 1){
            return 1;
        }

        Arrays.sort(nums);
        int count = 1;
        int max = 0;

        for(int i = 1; i < nums.length; i++){
            if(nums.length == 2 && nums[i] - nums[i-1] == 0){
                max = 1;
            }else if(nums[i] - nums[i-1] == 0){
                continue;
            }else if(nums[i] - nums[i-1] == 1){
                count++;
                if(count > max){
                    max = count;
                }
            }else{
                if(count > max){
                    max = count;
                }
                count = 1;
            }
        }
        return max;
    }
}

File: swdevelop24/Arrays_Hashing/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        nums.sort()
        cnt = 1
        ret = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                if nums[i] == (nums[i - 1] + 1):
                    cnt += 1
                    ret = max(cnt, ret)
                else:
                    cnt = 1

        """    
        if len(nums) == 0:
            ret =0
        """

        if not nums:
            ret = 0

        return ret


sol = Solution()
print(sol.longestConsecutive([]))
print(sol.longestConsecutive([100, 4, 200, 1, 3, 2]))
print(sol.longestConsecutive([0, 3, 7, 2, 5, 8, 4, 6, 0, 1]))
